### PR TITLE
Jest testing

### PR DESCRIPTION
**Pull request by hulien22**
Adding in jest tests and fixing up bugs found while creating the bugs.

### CODE REVIEW

**Comment by cognite-cicd** [comment]
[pr-bot]
Run this build with `docker run --rm -p 3000:3000 cognite/grafana-cdp-dev:pr-28`

**Comment by cognite-cicd** [comment]
[pr-bot]
Run this build with `docker run --rm -p 3000:3000 cognite/grafana-cdp-dev:pr-28`

File: src/spec/datasource.test.ts [hunk 1]
```diff
@@ -0,0 +1,1363 @@
...
+  return response;
+}
+
+describe('CogniteDatasource', () => {
+  const ctx: any = {};
+  const instanceSettings = {
+    id: 1,
+    name: 'Cognite Test Data',
+    type: 'cognitedata-platform-datasource',
+    url: '/api/datasources/proxy/6',
```
Comment (martincognite) [inline src/spec/datasource.test.ts]: I don't know what this `/6` is, is it deterministic?

Comment (hulien22) [reply src/spec/datasource.test.ts]: Grafana uses that url to distinguish between the different datasources that are included, this just happened to be one that I copied from one of my test datasources. Doesn't actually do anything in these tests.

**Pull request merged**